### ISSUE TITLE
Investigate gradient flow through FDTD

### ISSUE THREAD

**Issue by gomezzz** [esa/NIDN#53]
# Feature
## Desired Behavior / Functionality
We need the gradients for NN training

## What Needs to Be Done
See if they are there and if not where do we lose them

1. Set requires_grad variable on eps grid to tensor to true
2. Run compute spectrum 
3. Check if there are gradients on the outcoming spectrum
4. If not, go through code and check where they stop being present

(ofc everything has to be a tensor, so if it stops being a tensor inbetween the gradients are lost)

## How Can It Be Tested
use `print(variable.grad)` to print gradients (if available)
